offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -448,6 +452,34 @@ class Offer {
         this.availableVersions.addAll(copyIdentifier, this.identifier)
     }
 
+    /**
+     * This method returns the total discount amount that was applied the offer.
+     *
+     * It is expected to return the sum of all individual item discount amounts that have been
+     * determined by the number of samples an individual data analysis service has been provided for.
+     *
+     * @return the total discount amount applied in the offer
+     * @since 1.1.0
+     */
+    double getTotalDiscountAmount(){
+        return calculateTotalDiscountAmount()
+    }
+
+    private double calculateTotalDiscountAmount() {
+        return items.stream()
+                .map({it.quantityDiscount})
+                .reduce(0, (a,b)-> a + b)
+    }
```

Comment:
I do not understand this reduction. I think that using the `sum` would be clearer to read.
```suggestion
        return items.sum{ it.quantityDiscount }
    }
```

Reply:
That's nit isnt it? :D I would say that this highly depends on your personal preference, map reduce is a standard computer science idiom ;)

Reply:
I think improving readability is never wrong 😄 If you absolutely want to have your solution in the code this is fine, I would prefer the more explicit solution.